test/StardustProvider.spec.ts

```diff
@@ -0,0 +1,38 @@
+import test from 'japa';
+import { setup, teardown } from './utils';
+
+test.group('Server', (group) => {
+  group.afterEach(async () => {
+    await teardown();
+  });
+
+  test('Should handle empty router gracefully', async (assert) => {
+    const app = await setup();
+    const view = app.container.use('Adonis/Core/View');
+
+    view.registerTemplate('dummy', { template: '@routes()' });
+
+    assert.equal(
+      (await view.render('dummy')).trim(),
+      `<script>
+  window.stardust = {namedRoutes: {}};
+</script>`,
+    );
+  });
+
+  test.skip('Should render named routes', async (assert) => {
+    const app = await setup();
+    const router = app.container.use('Adonis/Core/Route');
+    const view = app.container.use('Adonis/Core/View');
+
+    router.get('/', async () => {}).as('index');
```

Comment:
@thetutlage this is driving me crazy. for some reason the routes don't register here. any suggestions on what i might be missing?

Reply:
Just call `router.commit` to process routes. 

Side note: The router basically creates a read optimised tree once you call `router.commit` so that the route lookup is quick during req/res lifecycle.

Reply:
thanks for responding. i am calling `route.commit` on line 30.
when i'm calling `route.toJSON()` in the test it seems like everything is there, but when I do the same in the provider it's empty. it's probable the way i set up the test.
i'd appreciate if you could take another look 🙏

Reply:
Yup, coz the `ready` method on the routes is called before any routes have been registered by the test. If you closely look at the codepath. This is how it looks 

```ts
test body
  -> call setup
       -> register providers, boot providers, call ready method
            -> ready method calls `route.toJSON()`
  -> now routes are registered
```

Basically, you need to remove the `app.start()` method from the utils file and call it after the routes have been registered.